lib/tasks/check.ex

```diff
@@ -31,7 +31,16 @@ defmodule Mix.Tasks.Workshop.Check do
 
   # when checking user solution
   defp handle_result(%Result{errors: [], warnings: []}) do
-    Session.get(:current_exercise) |> Exercise.set_status(:completed)
+    exercise = Session.get(:current_exercise)
+    exercise_module = Exercise.load(exercise)
+    identifier = Exercise.get_identifier(exercise_module)
+
+    unless Workshop.State.get(:exercises)[identifier][:status] == :completed do
```

Comment:
Yeah, I guess that getting the status of a given exercise should have a helper method by now; I seem to use it quite a few places :+1: